app/services/hyrax/workflow/workflow_importer.rb

```diff
@@ -60,25 +74,34 @@ def validate!
 
       public
 
+      attr_accessor :errors
+
       def call
+        self.errors = []
         Array.wrap(data.fetch(:workflows)).map do |configuration|
-          find_or_create_from(configuration: configuration)
+          begin
+            find_or_create_from(configuration: configuration)
+          rescue InvalidStateRemovalException => e
+            error = I18n.t('hyrax.workflow.load.state_error', workflow_name: e.state.workflow.name, state_name: e.state.name, entity_count: e.state.entities.count)
+            Rails.logger.error(error)
+            errors << error
+            Sipity::Workflow.find_by(name: configuration[:name])
+          end
         end
       end
 
       private
 
         def find_or_create_from(configuration:)
-          workflow = Sipity::Workflow.find_or_initialize_by(name: configuration.fetch(:name)) do |wf|
```

Comment:
@jeremyf It does not change anything unless the workflow fails becuase it removes a state it is not allowed.  At that point if you save changes to the workflow and then error later the workflow was half updated.  All the tests still passed when I moved the code around, but my tests for failing did not.  Do you have an example of your failure scenario?